### PR TITLE
Edits to UG and DG

### PR DESCRIPTION
**Pull request by meerkatboy**
Addresses the issues:
#308 - added tip on adding tags to company
#268 - added to Known Issues. Will add to Planned Enhancements in DG
#258 - created the relevant user stories
#251 - edited the caption to "update" application
#322 - added a baseline user story

Also made some minor edits with regards to punctuation and alphabet casing, after initial skimming through,

note: ignore the branch name, edited both DG and UG

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -405,26 +405,34 @@ methods of `UniqueCompanyList` that manipulates its `internalList`.
...
+| `* *`    | user                                                        | export my data out of InternBuddy                             | have a data backup and/or share company information with fellow peers                                    |
+| `* * *`  | STEM major planning to apply for internships                | find a specific company by name                               | retrieve information about a company without having to go through the entire list                        |
+| `* * *`  | STEM major planning to apply for internships                | find specific companies based on tags                         | retrieve information on companies attached to a certain tag without having to go through the entire list |
+| `* * *`  | STEM major planning to apply for internships                | find specific companies based on application names or details | retrieve information about companies without having to go through the entire list                        |
```
Comment (uniqly) [inline docs/DeveloperGuide.md:416]: This should be 2 stars

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -405,26 +405,34 @@ methods of `UniqueCompanyList` that manipulates its `internalList`.
...
+| `* *`    | STEM major planning to apply for internships                | update the information for a specific company                 | ensure my information is up-to-date                                                                      |
+| `*`      | STEM major planning to apply for internships                | add referral contacts to companies                            | remember to include them when applying                                                                   |
+| `* * *`  | STEM major planning to apply for internships                | favourite a company                                           | mark that company as a priority                                                                          |
+| `* * *`  | STEM major planning to apply for internships                | unfavourite a company                                         | unmark that company as a priority if no longer relevant                                                  |
```
Comment (uniqly) [inline docs/DeveloperGuide.md:421]: This should be 2 stars as well

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -405,26 +405,34 @@ methods of `UniqueCompanyList` that manipulates its `internalList`.
...
+| `*`      | STEM major currently applying for internships               | add notes for upcoming / completed interviews                 | find a specific piece of info for that interview in the future                                           |
+| `*`      | STEM major currently applying for internships               | organize interview timings                                    | ensure that they do not overlap                                                                          |
+| `* * *`  | STEM major planning to apply or is applying for internships | remove companies which are no longer relevant to me           | reduce clutter in my list of potential companies                                                         |
+| `* * *`  |                                                             |                                                               |                                                                                                          |
```
Comment (uniqly) [inline docs/DeveloperGuide.md:434]: Why is there an empty row?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -405,26 +405,34 @@ methods of `UniqueCompanyList` that manipulates its `internalList`.
...
+
 
 *{More to be added}*
```
Comment (uniqly) [inline docs/DeveloperGuide.md:437]: Can remove this.

File: docs/UserGuide.md [hunk 5]
```diff
@@ -243,7 +243,8 @@ Format: `add n/NAME e/EMAIL [p/PHONE_NUMBER] [a/ADDRESS] [t/TAG]…​`
 <box type="tip" seamless>
 
 **Tip 1:** A company can have any number of tags (including 0) <br>
-**Tip 2:** A company requires a name and email at the minimum
+**Tip 2:** Tags can be a useful way to categorise different companies into various groups!
```
Comment (uniqly) [inline docs/UserGuide.md:246]:
```suggestion
**Tip 2:** Tags can be a useful way to categorise different companies into various groups! <br>
```

**Review by uniqly — CHANGES_REQUESTED** [review]
Some edits to make, but looks good. @meerkatboy Will the adding of the planned enhancement be in a different PR #297 or will it be in this one

**Review by blanklogic — APPROVED** [review]
LGTM!

**Review by uniqly — APPROVED** [review]
LGTM!

**Pull request merged**